### DISCUSSION TITLE
Is there any way for queries to return the actual model instead of Model?

### DISCUSSION THREAD

**Discussion by makupi** [collerek/ormar#111] [Q&A] [answered]
I want to work with typehints, but unfortunately `UserModel.objects.get()` is typed to return Model and not UserModel.    
PyCharm now complains that I'm passing Model and not UserModel. Is there any way to get around this, besides making my typehints Model?

**Comment by collerek** [comment, marked as answer]
Yeah I noticed it some time ago and I want to update the types with Generics so it should work better. You can open an issue with types update so it won't be missed along other tasks.

As a immediate yet quite ugly solution you can cast the returned instance to proper type like this:
```python
from typing import cast
instance = UserModel.objects.get()
instance = cast(UserModel, instance)
```

**Reply by collerek** [reply]
Issue #112 is created to solve this issue

**Reply by collerek** [reply]
It was partially solved in 0.10.0. All `Model.objects.method()` should return the correct models and not the base `ormar.Model` type.
